src/Sentry/BuiltInSystemDiagnosticsMeters.cs

```diff
@@ -0,0 +1,170 @@
+
+    [GeneratedRegex(MicrosoftAspNetCoreHttpConnectionsPattern, RegexOptions.Compiled)]
+    private static partial Regex MicrosoftAspNetCoreHttpConnectionsRegex();
+#else
+    public static readonly SubstringOrRegexPattern MicrosoftAspNetCoreHttpConnections = new Regex(MicrosoftAspNetCoreHttpConnectionsPattern, RegexOptions.Compiled);
+#endif
+
+    /// <summary>
+    /// Matches the built in Microsoft.Extensions.Diagnostics.HealthChecks metrics
+    /// </summary>
+#if NET8_0_OR_GREATER
+    public static readonly SubstringOrRegexPattern MicrosoftExtensionsDiagnosticsHealthChecks = MicrosoftExtensionsDiagnosticsHealthChecksRegex();
+
+    [GeneratedRegex(MicrosoftExtensionsDiagnosticsHealthChecksPattern, RegexOptions.Compiled)]
+    private static partial Regex MicrosoftExtensionsDiagnosticsHealthChecksRegex();
+#else
+    public static readonly SubstringOrRegexPattern MicrosoftExtensionsDiagnosticsHealthChecks = new Regex(MicrosoftExtensionsDiagnosticsHealthChecksPattern, RegexOptions.Compiled);
+#endif
+
+    /// <summary>
+    /// Matches the built in Microsoft.Extensions.Diagnostics.ResourceMonitoring metrics
+    /// </summary>
+#if NET8_0_OR_GREATER
+    public static readonly SubstringOrRegexPattern MicrosoftExtensionsDiagnosticsResourceMonitoring = MicrosoftExtensionsDiagnosticsResourceMonitoringRegex();
+
+    [GeneratedRegex(MicrosoftExtensionsDiagnosticsResourceMonitoringPattern, RegexOptions.Compiled)]
+    private static partial Regex MicrosoftExtensionsDiagnosticsResourceMonitoringRegex();
+#else
+    public static readonly SubstringOrRegexPattern MicrosoftExtensionsDiagnosticsResourceMonitoring = new Regex(MicrosoftExtensionsDiagnosticsResourceMonitoringPattern, RegexOptions.Compiled);
+#endif
+
+    /// <summary>
+    /// Matches the built in System.Net.NameResolution metrics
+    /// </summary>
+#if NET8_0_OR_GREATER
+    public static readonly SubstringOrRegexPattern SystemNetNameResolution = SystemNetNameResolutionRegex();
+
+    [GeneratedRegex(SystemNetNameResolutionPattern, RegexOptions.Compiled)]
+    private static partial Regex SystemNetNameResolutionRegex();
+#else
+    public static readonly SubstringOrRegexPattern SystemNetNameResolution = new Regex(SystemNetNameResolutionPattern, RegexOptions.Compiled);
+#endif
+
+    /// <summary>
+    /// Matches the built in <see cref="System.Net.Http"/> metrics
+    /// </summary>
+#if NET8_0_OR_GREATER
+    public static readonly SubstringOrRegexPattern SystemNetHttp = SystemNetHttpRegex();
+
+    [GeneratedRegex(SystemNetHttpPattern, RegexOptions.Compiled)]
+    private static partial Regex SystemNetHttpRegex();
+#else
+    public static readonly SubstringOrRegexPattern SystemNetHttp = new Regex(SystemNetHttpPattern, RegexOptions.Compiled);
+#endif
+
+    /// <summary>
+    /// Matches all built in metrics
+    /// </summary>
+    /// <returns></returns>
+    public static IList<SubstringOrRegexPattern> All = [
```

Comment:
How about Lazy<T>?

I wonder if we can avoid allocating this list and all regex's if the user is not even setting the default metrics. Or if they are setting specific ones here instead of leaving the default All